internal/grpc/services/gateway/usershareprovider.go

```diff
@@ -106,6 +106,15 @@ func (s *svc) ListShares(ctx context.Context, req *collaboration.ListSharesReque
 }
 
 func (s *svc) updateShare(ctx context.Context, req *collaboration.UpdateShareRequest) (*collaboration.UpdateShareResponse, error) {
+	// TODO: update wopi server
+	// FIXME This is a workaround that should prevent removing or changing the share permissions when the file is locked.
+	// https://github.com/owncloud/ocis/issues/8474
+	if status, err := s.checkLock(ctx, req.GetShare().GetId()); err != nil {
+		return &collaboration.UpdateShareResponse{
+			Status: status,
+		}, nil
+	}
+
```

Comment:
I know I am a bit late to the party. But isn't this racy? The file  lock can  be created  after this check, before updating the share. Classical TOCTOU problem, isn't it?

Reply:
Right, we would have to send the lock in the opaque of the sharing requests to make this (more) *atomic*.